plugins/poll/lib/post_validator.rb

```diff
@@ -9,7 +9,7 @@ def initialize(post)
     def validate_post
       min_trust_level = SiteSetting.poll_minimum_trust_level_to_create
 
-      if @post&.user&.staff? || @post&.user&.trust_level >= TrustLevel[min_trust_level] || @post&.topic&.pm_with_non_human_user?
+      if @post&.acting_user&.staff? || @post&.acting_user&.trust_level >= TrustLevel[min_trust_level] || @post&.topic&.pm_with_non_human_user?
```

Comment:
Isn't this an old bug? not sure & is helping here at all

```
nil&.test >= 22
Traceback (most recent call last):
        4: from /home/sam/.rubies/ruby-2.7.5/bin/irb:23:in `<main>'
        3: from /home/sam/.rubies/ruby-2.7.5/bin/irb:23:in `load'
        2: from /home/sam/.rubies/ruby-2.7.5/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):5
NoMethodError (undefined method `>=' for nil:NilClass)
```

Reply:
That's a good observation. I do not think that happens very often in production because it rarely happens for `acting_user` to be `nil`. Also, I do not think that `@post` is ever `nil` either.